src/workers/race_event_status_worker.rs

```diff
@@ -96,44 +258,108 @@ pub async fn cron(mut sd: Receiver<Shutdown>, state: Arc<DiscordState>) {
     warn!("Race event worker quit");
 }
 
-async fn sync_race_status(state: &Arc<DiscordState>) -> Result<(), NMGLeagueBotError> {
+async fn sync_race_status(
+    state: &Arc<DiscordState>,
+    helper_bot: &Arc<HelperBot>,
+) -> Result<(), NMGLeagueBotError> {
     let mut conn_o = state.diesel_cxn().await?;
     let conn = conn_o.deref_mut();
-    let (race_infos, players) = get_season_race_info(conn)?;
-    let mut existing_events = get_existing_events_by_id(state).await?;
+    let race_infos = get_season_race_info(state, conn).await?;
+    let bri_ids = race_infos
+        .iter()
+        .map(|bundle| bundle.bri.get_id())
+        .collect::<Vec<_>>();
 
-    for mut bundle in race_infos {
-        let new_status = match get_event_content(&bundle, &players, state, conn).await {
-            Ok(status) => status,
-            Err(e) => {
-                warn!(
-                    "Error getting event content for race {}: {e}",
-                    bundle.race.id
-                );
-                continue;
-            }
-        };
-        let existing_event = if let Some(gse_id) = bundle.bri.get_scheduled_event_id() {
+    let mut race_events_by_guild = RaceEvent::get_for_bri_ids(&bri_ids, conn)?
+        .into_iter()
+        .group_by(|re| re.guild_id.clone())
+        .into_iter()
+        .map(|(guild_id, race_events)| {
+            (
+                guild_id,
+                race_events
+                    .map(|re| (re.bracket_race_info_id, re))
+                    .collect(),
+            )
+        })
+        .collect::<HashMap<_, _>>();
+
+    for gev in helper_bot.guild_event_configs() {
+        let race_events_by_bri_id = race_events_by_guild
+            .remove(&gev.guild_id.to_string())
+            .unwrap_or(HashMap::new());
+
+        if let Err(e) =
+            sync_events_in_a_guild(gev, helper_bot, &race_infos, race_events_by_bri_id, conn).await
+        {
+            warn!("Error syncing events in a guild: {e}");
+        }
+    }
+    Ok(())
+}
+
+async fn sync_events_in_a_guild(
+    guild_event_config: GuildEventConfig,
+    helper_bot: &Arc<HelperBot>,
+    race_infos: &[RaceInfoBundle],
+    mut race_events_by_bri_id: HashMap<i32, RaceEvent>,
+    conn: &mut SqliteConnection,
+) -> Result<(), NMGLeagueBotError> {
+    let mut existing_events =
+        get_existing_events_by_id(guild_event_config.guild_id, helper_bot).await?;
+
+    for bundle in race_infos {
+        let race_event = race_events_by_bri_id.remove(&bundle.bri.id);
```

Comment:
maybe we should check if this hashmap is empty at the end? it would be pretty weird if it weren't but idk what we'd do about it